kernel/src/main/scala/cats/kernel/instances/either.scala

```diff
@@ -63,21 +63,35 @@ trait EitherInstances0 extends EitherInstances1 {
           }
         }
     }
-}
 
-trait EitherInstances1 {
-  implicit def catsStdEqForEither[A, B](implicit A: Eq[A], B: Eq[B]): Eq[Either[A, B]] =
-    new Eq[Either[A, B]] {
-      def eqv(x: Either[A, B], y: Either[A, B]): Boolean =
+  implicit def catsStdHashForEither[A, B](implicit A: Hash[A], B: Hash[B]): Hash[Either[A, B]] =
+    new EitherEq[A, B](A, B) with Hash[Either[A, B]] {
+      def hash(x: Either[A, B]): Int = {
         x match {
-          case Left(xx) => y match {
-            case Left(yy) => A.eqv(xx, yy)
-            case Right(_) => false
-          }
-          case Right(xx) => y match {
-            case Left(_) => false
-            case Right(yy) => B.eqv(xx, yy)
-          }
+          case Left(xx) => Left(A.hash(xx)).##
```

Comment:
I want to have the default Hash instance for `Either[A, B]` perform the exact same way as the universal hash (##) given the Hash instance for `A` and `B` are universal hashes.
So maybe I'll copy the function implementation for case classes in Scala?
Basically I want `EitherHash[A, B](DefaultHashInstance[A], DefaultHashInstance[B])(Left(a)) == Left(a).##`.